src/sessions/base-session-handler.ts

```diff
@@ -206,14 +206,16 @@ export default abstract class BaseSessionHandler {
 
     await Promise.all(destroyMediaPromises);
 
-    if (!stream) {
+    const newStartMediaContraints = {
+      audio: options.audioDeviceId,
+      /* if video is muted, we don't want to request it */
+      video: !session.videoMuted && options.videoDeviceId,
+      session
+    };
+
+    if (!stream && (newStartMediaContraints.audio || newStartMediaContraints.video)) {
```

Comment:
I think this `if` check needs to be a bit more verbose to accommodate the `null` option (which the sdk interprets as "system default device" and eventually just gets passed to `getUserMedia` (gUM) as `true`). 
``` ts
if (!stream && 
  (newStartMediaContraints.audio !== undefined  || 
  newStartMediaContraints.video !== undefined)
) { ... 
```
If you want to read more about how we implemented the options, you can see the docs for [IMediaRequestOptions](https://github.com/MyPureCloud/genesys-cloud-webrtc-sdk/blob/v6-beta/doc/media.md#imediarequestoptions) and [IUpdateOutgoingMedia](https://github.com/MyPureCloud/genesys-cloud-webrtc-sdk/blob/v6-beta/doc/media.md#iupdateoutgoingmedia).

Reply:
Yeah you're right, I'll go ahead and make the changes. Good catch.